flocx_market/db/sqlalchemy/offer_api.py

```diff
@@ -0,0 +1,41 @@
+from flocx_market.db.orm import orm
+from flocx_market.db.sqlalchemy.offer_model import OfferModel
+from uuid import uuid4
+import json
+
+
+class OfferApi(OfferModel):
+    def __init__(self, provider_id, creator_id, marketplace_date_created, status, server_id, start_time, end_time, server_config, cost):
+        self.marketplace_offer_id = str(uuid4())
+        self.provider_id = provider_id
+        self.creator_id = creator_id
+        self.marketplace_date_created = marketplace_date_created
+        self.status = status
+        self.server_id = server_id
+        self.start_time = start_time
+        self.end_time = end_time
+        self.server_config = json.dumps(server_config)
+        self.cost = cost
+
+    def as_dict(self):
+        d = {}
+        for col in self.__table__.columns:
+            if col.name in ['marketplace_date_created', 'start_time', 'end_time']:
+                d[col.name] = getattr(self, col.name).isoformat()
+            elif col.name in ['server_config']:
+                d[col.name] = json.loads(getattr(self, col.name))
```

Comment:
I would really like to see this fixed before we merge these changes: the data model should return a Python dictionary, not a JSON string. The format in which the data is stored in the database should be transparent to client code. If we want to stop using JSON and use something else, it should *only* require changes to the data model and nowhere else. Any error handling relating to JSON serialization/deserialization should be handled by the data model, rather than requiring client code to handle that.